src/qt/ve_qitems_mqtt.cpp

```diff
@@ -0,0 +1,477 @@
+#include <veutil/qt/ve_qitems_mqtt.hpp>
+
+#include <QRandomGenerator>
+#include <QJsonDocument>
+#include <QJsonObject>
+
+//--
+
+VeQItemMqtt::VeQItemMqtt(VeQItemMqttProducer *producer)
+	: VeQItem(producer, nullptr)
+{
+	connect(producer, &VeQItemMqttProducer::connectionStateChanged,
+		this, [this] {
+			const VeQItemMqttProducer::ConnectionState state = mqttProducer()->connectionState();
+			switch (state) {
+			case VeQItemMqttProducer::Ready:
+				// TODO: may not be required, since we subscribe to everything
+				//       and send keepalive to read all values when Ready.
+				mqttProducer()->readValue(uniqueId());
+				break;
+			case VeQItemMqttProducer::Disconnected:
+				produceValue(QVariant(), VeQItem::Offline);
+				break;
+			default: break;
+			}
+		});
+}
+
+int VeQItemMqtt::setValue(QVariant const &value)
+{
+	if (!mqttProducer() || !mqttProducer()->publishValue(uniqueId(), value)) {
+		return -1;
```

Comment:
I assume there is some precedent for returning -1 for failure, 0  for success. Otherwise I would just use true/false

Reply:
Yes, check the header, it's an override of a virtual function from VeQItem.